### PR TITLE
Bound page size and eliminate wasteful RSC round-trips on table interactions

### PR DESCRIPTION
**Pull request by TomEbeyer**
## Description

Table responsiveness work, in two areas — a page-size guard, and cutting unnecessary work per interaction (both server round-trips and client re-renders). The client-side items were found by profiling the homepage demo (React `<Profiler>` commit counts + row-render counters); dev-mode ms are inflated but the commit/render **counts** are exact.

### Server round-trips per interaction (homepage demo)

| Interaction | Before | After |
|---|--:|--:|
| Sort | 1 | 1 *(needed)* |
| Paginate | 1 | 1 *(needed)* |
| **Select a row** | **1** | **0** |
| **Select a row (1st, clean URL)** | **1** | **0** |
| Column show/hide, reorder | 1 | 0 |

### Client re-renders per interaction (row re-renders, 10-row page; dev, StrictMode-doubled)

| Interaction | Before | After |
|---|--:|--:|
| Select a row (1st, clean URL) | **62** | **2** |
| Apply filter | **40** | **20** |
| Sort / Paginate | 20 | 20 *(1 logical render; the 2× is dev StrictMode, not prod)* |

## Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ Performance improvement
- [x] 🔒 Security fix
- [x] ✅ Tests (adding or updating tests)

## Package(s) Affected

- [x] `@better-tables/core`
- [x] `@better-tables/ui`
- [x] `@better-tables/adapters-drizzle`
- [x] Documentation (marketing site demo)

### 1. Page-size guard — `@better-tables/core` + `@better-tables/adapters-drizzle`
- Add `DEFAULT_MAX_PAGE_SIZE` (200) in core.
- Add a `maxPageSize` adapter option (default `DEFAULT_MAX_PAGE_SIZE`; non-positive disables). `fetchData` clamps an over-cap `pagination.limit` before querying and reflects the clamped value in returned pagination metadata. Never mutates the caller's params. `pagination.limit` is commonly URL-driven, so this closes an unbounded-query vector and a large-client-render source. New unit tests.

### 2. Avoid wasteful RSC round-trips — `@better-tables/ui` + marketing demo
- `useTableUrlSync`: skip the URL write-out entirely when every serialized param already equals the current URL. Changes that touch no synced param (row selection is the common one) no longer call `setParams`, so adapters that navigate on `setParams` don't refetch for nothing. Benefits every consumer/adapter.
- `useNextjsUrlAdapter.setParams` (demo): no-op when the resulting query is unchanged; `router.replace` (not `push`) only when a server-affecting param (page/limit/filters/sorting) actually changed value; otherwise a shallow `history.replaceState` for client-only view state (column visibility/order), keeping it shareable without a server round-trip.

### 3. Cut client re-renders — `@better-tables/ui`
- `useTableUrlSync`: don't introduce default `page`/`limit` into a URL that doesn't already carry them. Capture the table's starting pagination and drop `page`/`limit` from a write when they're at that default and absent from the URL. Previously the **first** interaction on a clean URL (e.g. a selection) serialized the default `page=1&limit=<default>`, which differed from the empty URL and triggered a navigation + refetch + full re-render cascade (**62 row re-renders → 2**).
- `autoShowFilteredColumns`: only call `setColumnVisibility` when visibility actually changed. Filtering an already-visible column (the common case) otherwise pushed a new-but-equal object into the store and re-rendered the whole table (**filter-apply row re-renders ~halved**).

## Testing

- [x] I have added/updated tests for my changes
- [x] All existing tests pass
- [x] I have tested manually in the following scenarios:
  - [x] Basic functionality
  - [x] Edge cases
  - [x] Browser compatibility (profiled in a real Chromium)

### Test Results

```
# adapter page-size clamp (new) + existing SQLite suite
34 pass, 0 fail (packages/adapters/drizzle)

# In-browser (Playwright, homepage demo)
RSC round-trips: sort=1 paginate=1 select-row=0 (incl. 1st on clean URL) column-toggle=0
  - "skip unchanged" core fix proven independently with a naive always-push adapter: select 1 -> 0
Client re-renders (row): first-selection 62 -> 2 ; filter-apply 40 -> 20
Functional: sort reorders + writes sorting= ; selection checks box, no URL change, no cascade ;
  sort works after selection (router not wedged) ; paginate updates + page= ; ?limit=1000 -> 200 rows
```

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Additional Notes

A changeset is included for the published-package changes (`@better-tables/core`, `@better-tables/adapters-drizzle`). `@better-tables/ui` is in the changeset ignore list (distributed via CLI copy, not npm), so it needs none.

Profiling also flagged a "double render" on data changes, but that turned out to be **React StrictMode in dev** (every render doubled), not a production issue — so no change there. A `setTotal`-guard optimization was prototyped and dropped because it had no measurable impact on the profiled interactions (`totalCount` is stable, so the effect doesn't re-fire `setTotal` with the same value).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/adapters/drizzle/src/drizzle-adapter.ts [hunk 1]
```diff
@@ -458,6 +476,13 @@ export class DrizzleAdapter<TSchema extends Record<string, unknown>, TDriver ext
   ): Promise<FetchDataResult<InferSelectModelFromFilteredSchema<TSchema>>> {
     const startTime = Date.now();
 
+    // Clamp an oversized page size (commonly URL-driven via `?limit=…`) before
+    // it reaches the database, so a crafted request can't pull an unbounded
+    // result set or force the client to render tens of thousands of rows. The
+    // clamped value flows through to both the query and the returned pagination
+    // metadata below, since everything downstream reads `params.pagination`.
+    params = this.withClampedPageSize(params);
```
[the remark this answers is not in the document]
Comment (TomEbeyer) [reply packages/adapters/drizzle/src/drizzle-adapter.ts]: Good catch — fixed in caec2b8. `fetchData` now takes an internal `clampPageSize` flag and `exportData` passes `clampPageSize: false`, so the export keeps its own `maxRows`/`DEFAULT_EXPORT_ROW_CAP` bound and is never truncated to the interactive `maxPageSize`. Added a regression test that seeds >200 rows and asserts `exportData` returns them all.

Same commit also addresses the other review points:
- **Fractional cap (P2):** the cap is floored to a whole number, so `maxPageSize: 2.5` → 2 (test added).
- **`tableId` change (P2):** `useTableUrlSync` now re-captures the default pagination when `tableId` changes, so table B's writes aren't measured against table A's defaults.
- **Weak clamp test (P3):** added a test that seeds >`DEFAULT_MAX_PAGE_SIZE` rows and asserts the SQL result set is actually limited (not just the metadata).
- **Hook tests:** added coverage for not introducing default `page`/`limit` into a clean URL, and for suppressing no-op writes.
- **CI:** the failure was an unrelated formatting error in the new clamp test file (`biome check`), now formatted.

---
_Generated by [Claude Code](https://claude.ai/code)_
